File: 70-climbing-stairs/70-climbing-stairs.cpp
class Solution {
    vector<int> dp;
public:
    int climbStairs(int n) {
        dp.assign(n+1, -1);
        return fun(n);
    }
private:
    int fun(int n) {
        if (n == 0) 
            return 0;
        if (n == 1)
            return 1;
        if (n == 2)
            return 2;
        if (dp[n] != -1) 
            return dp[n];
        return dp[n] =  fun(n-1) + fun(n-2);
    }
};

File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        @cache
        def jump(i):
            if i == n-1:
                return 1
            if i == n-2:
                return 2
            one = jump(i+1)
            two = jump(i+2)
            return one + two
        return jump(0)
        
        
        
        
        
        

        
        
        
        
        